### PR TITLE
Email Preview Modal

### PR DESCRIPTION
**Pull request by djnunez-aot**
*Issue #1017:*
https://github.com/bcgov/met-public/issues/1017

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-web/src/components/comments/admin/review/CommentReview.tsx [hunk 1]
```diff
@@ -168,6 +187,25 @@ const CommentReview = () => {
     const defaultVerdict = comment_status_id !== CommentStatus.Pending ? comment_status_id : CommentStatus.Approved;
     return (
         <MetPageGridContainer>
+            <EmailPreviewModal
+                open={open}
+                handleClose={() => setOpen(false)}
+                header={'Your comment on (Engagement name) needs to be edited'}
+                emailText={[
```
Comment (jadmsaadaot) [inline met-web/src/components/comments/admin/review/CommentReview.tsx:194]: could we discuss this pattern in a meeting. I suggest we emulate Material UIs pattern for customizable rendering like renderInput and renderOption below in AutoComplete
image

**Review by jadmsaadaot — CHANGES_REQUESTED** [review]
sugessting to hold merging this until we discuss emailText pattern. We could change it to something like renderOption in the AutoComplete of a Material UI

**Review by tiagograf-AOT — DISMISSED** [review]
it looks pretty good i think you just missed a field in the template

File: met-web/src/components/comments/admin/review/emailPreview/EmailTemplates.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            </Grid>
+        </When>
+        <When condition={otherReason}>
+            <Grid item xs={12}>
+                <MetBody
+                    sx={{ mb: 1 }}
+                >{`One or many of your comments can't be published because ${otherReason}.`}</MetBody>
+            </Grid>
+        </When>
+    </>
```
[the remark this answers is not in the document]
Comment (djnunez-aot) [reply met-web/src/components/comments/admin/review/emailPreview/EmailTemplates.tsx:52]: perfect

[the remark this answers is not in the document]
Comment (djnunez-aot) [reply met-web/src/components/comments/admin/review/emailPreview/EmailTemplates.tsx:52]: ill fix that up right now

File: met-web/src/components/comments/admin/review/CommentReview.tsx [hunk 3]
```diff
@@ -47,10 +50,29 @@ const CommentReview = () => {
...
     const [updatedStaffNote, setUpdatedStaffNote] = useState<StaffNote[]>([]);
-
+    const [open, setOpen] = useState(false);
     const dispatch = useAppDispatch();
     const navigate = useNavigate();
     const { submissionId } = useParams();
+    const reviewNotes = updatedStaffNote.filter((staffNote) => staffNote.note_type == StaffNoteType.Review);
+    const internalNotes = updatedStaffNote.filter((staffNote) => staffNote.note_type == StaffNoteType.Internal);
+
+    const getEmailPreview = () => {
```
Comment (jadmsaadaot) [inline met-web/src/components/comments/admin/review/CommentReview.tsx:60]: This is now a functional component and should follow the pattern and the naming convention of functional components. Not a normal JS function

**Review by jadmsaadaot — APPROVED** [review]

File: met-web/src/components/comments/admin/review/CommentReview.tsx [hunk 4]
```diff
@@ -47,10 +50,29 @@ const CommentReview = () => {
     const [notifyEmail, setNotifyEmail] = useState(true);
     const [staffNote, setStaffNote] = useState<StaffNote[]>([]);
     const [updatedStaffNote, setUpdatedStaffNote] = useState<StaffNote[]>([]);
-
+    const [open, setOpen] = useState(false);
```
[the remark this answers is not in the document]
Comment (djnunez-aot) [reply met-web/src/components/comments/admin/review/CommentReview.tsx]: good idea changed that up

**Review by saravanpa-aot — APPROVED** [review]

**Pull request merged**